### PR TITLE
[doc] Tidy up Tor doc (more stringent)

### PR DESCRIPTION
**Pull request by ghost**
This is a follow up to https://github.com/bitcoin/bitcoin/pull/19638 that left two deprecated "hidden service/server" naming occurences.

It also shall make the chapter titles regarding creation of onion services stringent and easy to read and distinguish.

It removes the one and only reference to the testnet (here the testnet onion service port), as it is not explained that it references to the testnet and I do not know why it is mentioned there. It is only confusing. Also, as said, the testnet is not referenced at any other place in this document.

### CODE REVIEW

**Comment by practicalswift** [comment]
ACK 32045bbfd5d77513efc162be8d4e24ea67539e27

Weird that we were suggesting that users should open up port 18334 (in addition to 8334). Glad to see that addressed :)

**Review by RiccardoMasutti — APPROVED** [review]
ACK 32045bb

Thanks, it was an oversight

**Comment by laanwj** [comment]
Review ACK 32045bbfd5d77513efc162be8d4e24ea67539e27

File: doc/tor.md [hunk 1]
```diff
@@ -46,7 +46,6 @@ versions of Tor see [Section 3](#3-automatically-listen-on-tor).*
 
 	HiddenServiceDir /var/lib/tor/bitcoin-service/
 	HiddenServicePort 8333 127.0.0.1:8334
-	HiddenServicePort 18333 127.0.0.1:18334
```
Comment (laanwj) [inline doc/tor.md:49]: Agree re: testnet. if people want to run Tor testnet (or signet) nodes, they tend to be developers who know how to change around the ports here. It's just confusing as it is.

**Comment by wodry** [comment]
Thanks for reviewing, glad I could contribute for the better! :tada:

**Pull request merged**